grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy

```diff
@@ -94,38 +94,89 @@ class IndicatorDataService {
         return graphData
     }
 
-    GraphData getFillRate() {
-        List listData = []
-        List bar2Data = []
+    GraphData getFillRate(Location location, def destination, def params) {
+        Integer querySize = params.querySize ? params.querySize.toInteger() - 1 : 7
+    
+        if(destination == null) destination = location
         List listLabel = []
+        
+
+        List averageFillRateResult = []
+        List requestLinesSubmittedResult = []
+        List linesCancelledStockoutResult = []
+        List averageTargetFillRate = [0.9]*querySize
+
         Date today = new Date()
         today.clearTime()
-        for (int i = 5; i >= 0; i--) {
+        for (int i = querySize; i >= 0; i--) {
             def monthBegin = today.clone()
             def monthEnd = today.clone()
             monthBegin.set(month: today.month - i, date: 1)
             monthEnd.set(month: today.month - i + 1, date: 1)
+            
+            String monthLabel = new java.text.DateFormatSymbols().months[monthBegin.month]
 
-            def query1 = Requisition.executeQuery("""select count(*) from RequisitionItem where dateCreated >= ? and dateCreated < ?""", [monthBegin, monthEnd]);
+            listLabel.push("${monthLabel} ${monthBegin.year + 1900}")
 
-            def query2 = Requisition.executeQuery("""select count(*) from RequisitionItem where dateCreated >= ? and dateCreated < ? and quantityCanceled > 0 and (cancelReasonCode = 'STOCKOUT' or cancelReasonCode = 'LOW_STOCK' or cancelReasonCode = 'COULD_NOT_LOCATE')""", [monthBegin, monthEnd]);
-            String monthLabel = new java.text.DateFormatSymbols().months[monthBegin.month]
+            def averageFillRate = dataService.executeQuery("""
```

Comment:
Locally I am getting here Internal server error with message:`SELECT command denied to user 'openboxes_test'@'localhost' for table 'fill_rate'`. Should there be a migration with this table?

Reply:
Yes, the migration hasn't been done yet, you have to create the view locally : 

```
create or replace view fill_rate as 
SELECT 
r.id, 
r.status, 
ri.quantity, 
ri.quantity_canceled, 
ri.cancel_reason_code, 
ri.requisition_item_type, 
product.id as product_id,
product.product_code,
product.name as product_name,
t.transaction_date, month(t.transaction_date) as month, 
year(t.transaction_date) as year,
origin.id as origin_id, 
origin.name as origin_name,
destination.id as destination_id,
destination.name as destination_name,
case
when 
il.status = 'FORMULARY' then true
else false
end as formulary,
case
when 
ri.quantity_canceled > 0 
and ri.cancel_reason_code in ('STOCKOUT','LOW_STOCK','COULD_NOT_LOCATE') then 0 
else 1
end as fill_rate
FROM requisition as r 
join requisition_item ri on r.id = ri.requisition_id
join location as origin on origin.id = r.origin_id
join location as destination on destination.id = r.destination_id
join product as product on product.id = ri.product_id
left outer join transaction as t on t.requisition_id = r.id
left outer join inventory_level as il on ( il.inventory_id = origin.inventory_id and il.product_id = product.id )
where ri.requisition_item_type = 'ORIGINAL'
and r.status = 'ISSUED';
```

Reply:
ok

Reply:
Then I am adding 'needs-migration' label
cc @jmiranda